File: backend/app/schemas/spending.py
from pydantic import BaseModel


class CategoryResponse(BaseModel):
    name: str
    total: float
    count: int
    percentage: float


class SpendingSummaryResponse(BaseModel):
    total_spent: float
    transaction_count: int
    category_count: int
    categories: list[CategoryResponse]
    uncategorized_count: int
    uncategorized_percentage: float
    refund_total: float
    refund_count: int

File: frontend/src/types/spending.ts
export interface CategoryData {
  name: string;
  total: number;
  count: number;
  percentage: number;
}

export interface SpendingSummaryData {
  total_spent: number;
  transaction_count: number;
  category_count: number;
  categories: CategoryData[];
  uncategorized_count: number;
  uncategorized_percentage: number;
  refund_total: number;
  refund_count: number;
}
